test/integration/crud/explain.test.ts

```diff
@@ -0,0 +1,282 @@
+  type MongoClient,
+  MongoServerError
+} from '../../mongodb';
+
+describe('Explain', function () {
+  let client: MongoClient;
+  let db: Db;
+  let collection: Collection;
+  let commandsStarted: CommandStartedEvent[];
+
+  beforeEach(async function () {
+    client = this.configuration.newClient({ monitorCommands: true });
+    db = client.db('queryPlannerExplainResult');
+    collection = db.collection('test');
+    commandsStarted = [];
+
+    await collection.insertOne({ a: 1 });
+    client.on('commandStarted', event => commandsStarted.push(event));
+  });
+
+  afterEach(async function () {
+    await collection.drop();
+    await client.close();
+    commandsStarted = [];
+  });
+
+  context('when explain is set to true', () => {
+    it('deleteOne returns queryPlanner explain result', async function () {
+      const explanation = await collection.deleteOne({ a: 1 }, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+      expect(commandsStarted[0]).to.have.nested.property('command.explain.delete');
+    });
+
+    it('deleteMany returns queryPlanner explain result', async function () {
+      const explanation = await collection.deleteMany({ a: 1 }, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('updateOne returns queryPlanner explain result', async function () {
+      const explanation = await collection.updateOne(
+        { a: 1 },
+        { $inc: { a: 2 } },
+        { explain: true }
+      );
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('updateMany returns queryPlanner explain result', async function () {
+      const explanation = await collection.updateMany(
+        { a: 1 },
+        { $inc: { a: 2 } },
+        { explain: true }
+      );
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('distinct returns queryPlanner explain result', async function () {
+      const explanation = await collection.distinct('a', {}, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('findOneAndDelete returns queryPlanner explain result', async function () {
+      const explanation = await collection.findOneAndDelete({ a: 1 }, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('allPlansExecution returns verbose queryPlanner explain result', async function () {
+      const explanation = await collection.deleteOne({ a: 1 }, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+      expect(explanation).nested.property('executionStats.allPlansExecution').to.exist;
+    });
+
+    it('findOne returns queryPlanner explain result', async function () {
+      const explanation = await collection.findOne({ a: 1 }, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+  });
+
+  context('when explain is not set to true', () => {
```

Comment:
You can get rid of this context block since the value of `explain` is specified in each of the blocks inside this one